src/main/java/com/khmall/domain/category/CategoryValidator.java

```diff
@@ -0,0 +1,37 @@
+package com.khmall.domain.category;
+
+import static com.khmall.common.constants.CategoryConstants.DEPTH_EXCEEDED;
+import static com.khmall.common.constants.CategoryConstants.DEPTH_MAX;
+import static com.khmall.common.constants.CategoryConstants.NAME_DUPLICATE;
+
+import com.khmall.exception.custom.BadRequestException;
+import com.khmall.exception.custom.DuplicateException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CategoryValidator {
+  private final CategoryRepository categoryRepository;
+
+  // 깊이 제한 검증
+  public void validateDepth(Category parent) {
+    int depth = 1;
+    Category current = parent;
+    while (current != null) {
+      current = current.getParent();
+      depth++;
+      if (depth > DEPTH_MAX) {
+        throw new BadRequestException(DEPTH_EXCEEDED);
+      }
+    }
+  }
+
+  // 이름 중복 검증
+  public void validateDuplicateName(Category parent, String name) {
+    boolean exists = categoryRepository.existsByParentAndName(parent, name);
+    if (exists) {
+      throw new DuplicateException(NAME_DUPLICATE);
+    }
```

Comment:
```suggestion
    if (categoryRepository.existsByParentAndName(parent, name)) {
      throw new DuplicateException(NAME_DUPLICATE);
    }
```
불필요한 exist는 선언하지 않아도 될것 같습니다. 같은 depth의 같은 parent가진 category안에서만 name이 중복되지 않으면 되는 것으로 이해했구요.